src/clj/game/cards/operations.clj

```diff
@@ -322,6 +322,26 @@
    {:events {:pre-damage {:req (req (= target :brain)) :msg "do 1 additional brain damage"
                           :once :per-turn :effect (effect (damage-bonus :brain 1))}}}
 
+   "Distract the Masses"
+   (let [shuffle-two {:delayed-completion true
+                      :effect (effect (rfg-and-shuffle-rd-effect (find-cid (:cid card) (:discard corp)) 2))}
+         trash-from-hq {:delayed-completion true
+                        :prompt "Select up to 2 cards in HQ to trash"
+                        :choices {:max 2
+                                  :req #(and (= (:side %) "Corp")
+                                             (in-hand? %))}
+                        :msg (msg "trash " (quantify (count targets) "card") " from HQ")
+                        :effect (req (when-completed
+                                       (trash-cards state side eid targets nil)
```

Comment:
Don't pass an `eid` to a function if you're using it in `when-completed`; the macro will automatically generate a new eid to associate with the command. Right now DtM is being marked complete as soon as `trash-cards` finishes, because you're handing over DtM's eid to trash-cards. This would only be noticeable with Subcontract or AccDiag, but it's there.

Reply:
Makes sense, I had a feeling it was the wrong thing to do, but I kept getting arity issues and the only thing that seemed to work was passing through the eid.  Still not sure why passing a nil at the end calls the `trash-cards` function that receives arguments `[state side cards]`, need to read up more on Closure.

Reply:
It doesn't! 😄  `when-completed`, as a macro, will auto-insert an `eid` parameter that it generates itself as the third parameter if none is present, so you are indeed calling the 5-parameter `trash-cards` overload. The difference compared to your earlier approach is that `when-completed` will create a new unique `eid` intead of using the UI already assigned to the "current" ability.